components/Modals/EquipItemModal.tsx

```diff
@@ -53,312 +34,212 @@ export const EquipItemModal: React.FC = () => {
+          );
       }
-    },
-    [
-      publicClient,
-      game,
-      reloadGame,
-      character,
-      renderError,
-      selectedItem,
-      walletClient,
-      isEquipped,
-    ],
-  );
 
-  const content = () => {
-    if (txFailed) {
-      return (
-        <VStack py={10} spacing={4}>
-          <Text>Transaction failed.</Text>
-          <Button onClick={equipItemModal?.onClose} variant="outline">
-            Close
-          </Button>
-        </VStack>
+      const transactionhash = await executeAsCharacter(
+        character,
+        walletClient,
+        {
+          chain: walletClient.chain,
+          account: walletClient.account?.address as Address,
+          address: game.id as Address,
+          abi: parseAbi([
+            'function unequipItemFromCharacter(uint256 characterId, uint256 tokenId) external',
+            'function equipItemToCharacter(uint256 characterId, uint256 tokenId) external',
+          ]),
+          functionName: isEquipped
+            ? 'unequipItemFromCharacter'
+            : 'equipItemToCharacter',
+          args: [BigInt(character.characterId), BigInt(selectedItem.itemId)],
+        },
       );
+      return transactionhash;
+    } catch (e) {
+      throw e;
+    } finally {
+      setIsLoading(false);
     }
+  }, [game, character, selectedItem, walletClient, isEquipped]);
 
-    if (isSynced) {
-      return (
-        <VStack py={10} spacing={4}>
-          <Text>
-            {/* NOTE: Becuase the game data reloads before this text appears, the logic is flipped */}
-            Item(s) successfully {isEquipped ? 'unequipped' : 'equipped'}!
-          </Text>
-          <Button onClick={equipItemModal?.onClose} variant="outline">
-            Close
-          </Button>
-        </VStack>
-      );
-    }
-
-    if (txHash && selectedItem) {
-      return (
-        <TransactionPending
-          isSyncing={isSyncing}
-          text={`${isEquipped ? 'Unequipping' : 'Equipping'} ${
-            selectedItem.name
-          }.`}
-          txHash={txHash}
-          chainId={game?.chainId}
-        />
-      );
-    }
-
-    if (!character) {
-      return (
-        <VStack py={10} spacing={4}>
-          <Text>Character not found.</Text>
-          <Button onClick={equipItemModal?.onClose} variant="outline">
-            Close
-          </Button>
-        </VStack>
-      );
-    }
+  const heldItem = character?.heldItems.find(
+    e => e.itemId === selectedItem?.itemId,
+  );
 
-    const heldItem = character.heldItems.find(
-      e => e.itemId === selectedItem?.itemId,
-    );
+  const error = !character
+    ? 'Character not found'
+    : !selectedItem || !heldItem
+    ? 'Item not found'
+    : null;
 
-    if (!selectedItem || !heldItem) {
-      return (
+  return (
+    <ActionModal
+      {...{
+        isOpen: equipItemModal?.isOpen,
+        onClose: equipItemModal?.onClose,
+        header: `${isEquipped ? 'Unequip' : 'Equip'} Item`,
```

Comment:
This displays the wrong message after the tx is complete and the game reloads... wonder if we should just change this to 'equip / unequip item"

Reply:
Tweaked in 2c67903057c72ab57e7138a32fd90d7d01dc698e